client/templates/study_groups/study_group_settings.js

```diff
@@ -89,8 +103,9 @@ Template.studyGroupSettings.events({
             return Bert.alert( error.reason, 'danger', 'growl-top-right' );
           }
           if(result){
-            FlowRouter.go("all study groups");
-            return Bert.alert( 'Study Group transferred', 'success', 'growl-top-right' );
+            // Select the first tab in the study group
+            $('div.study-group-body > ul li:first-child > a').tab('show');
```

Comment:
` $('div.study-group-body > ul li:first-child > a')` seems a little crazy - can it be cleaned up for better readability?

Probably adding a `.study-group-tab` class on those links and querying it that way?

```javascript
$('.study-group-tab').first().show();
```

Reply:
@angelocordon  Your expression does not work.  
I change it to a more readable one.

Reply:
Here's a possible middle ground between complexity and hard coding:

```javascript
$('div.study-group-body [role=tablist] a:first').tab('show')
```

Reply:
@railsstudent curious to hear why it didn't work? It may require a couple more lines of code, but in the end, I think pushing for code readability will level up our code quality and help other contributors be able to understand our codebase better and raise our standards (which is always a good thing!). 

I'm not familiar with how we're using `.tab()` so that may be it - 

If you add class names for the links you can use that as selectors directly.

```html
<div class="study-group-body">
  <ul>
    <li>
       <a class="study-group-tab">...</a>
       <a class="study-group-tab">...</a>
         ...
</div>
```

```javascript
$('.study-group-tab').first().tab('show')
```

The HTML snippet above is a bit of a pseudo code just to show what I'm talking about. If this template is in the `staging` branch I can dig deeper. 

In any case, what do you guys think? Interested in hearing your thoughts - does this make it clearer/sense?

cc: @ajparise

Reply:
@angelocordon I'm of the opinion adding class selectors makes it more confusing and harder to maintain. Those classes serve no purpose other than reducing the complexity of a selector.

`$('div.study-group-body > ul li:first-child > a')` is precise and only selects exactly what it's supposed to. I think that actually improves the readability.

Reply:
Sad to hear my code quality is not good.  
I will take a step back to focus on learning of web development and let others tackle the issues.

Reply:
@ajparise I hear what you're saying, but I don't fully agree 😃 

I'm open to your opinions, however I think adding the class names to those links does exactly what you're saying - if we query the links via `$('.study-group-tab')`, jQuery traverses the DOM specifically for those elements and only those elements. 

When we pass `div.study-group-body > ul li:first-child > a` as a query selector, it works from right to left; it will first collect all `a` elements, then it will find the `a` elements within a `li` then `ul`, and then scope it down to `div.study-group-body`. 

Vs when we tell it to look specifically for an element with a specific class name, then it only needs to traverse the DOM once 😄 

Re: readability, perhaps `study-group-tab` does not have the correct semantics; `nav-tab` may be better? I'd argue that `$('.nav-tab').first()` is definitely more understandable. 

Alternatively, looks like we can also use `a[role="tab"]`. There are some slight performance differences regarding querying elements with classes vs attributes, but I guess they don't matter as much in 2017. 

```javascript
$('a[role="tab"]').first().tab('show')
```

or 

```javascript
// requires you to add a `nav-tab` class to the anchor links
$('.nav-tab').first().tab('show')
```

Either of the above does what you want @railsstudent.

Reply:
@railsstudent - please don't feel discouraged by all of this, @distalx gave some great feedback and this discussion between @ajparise and I is interesting in a sense that I hope you can find some insights from all of our perspectives. 

It's perfectly okay to take a step back, but also do it so you can get a better scope of the feature you're working on :)